addons/event_sms/models/event_mail.py

```diff
@@ -37,6 +38,13 @@ def _compute_template_model_id(self):
         sms_mails.template_model_id = sms_model
         super(EventMailScheduler, self - sms_mails)._compute_template_model_id()
 
+    @api.constrains('notification_type', 'template_ref')
+    def _check_template_ref(self):
+        """If the notification type is "sms", then the model should be "sms.template"."""
```

Comment:
Can we just check all of them? I think it's technically possible to select any one model with any other one currently.